modules/katello_lifecycle_environment.py

```diff
@@ -0,0 +1,198 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: katello_lifecycle_environment
+short_description: Create and Manage Katello lifecycle environments
+description:
+    - Create and Manage Katello lifecycle environments
+author: "Andrew Kofink (@akofink)"
+requirements:
+    - "nailgun >= 0.28.0"
+    - "python >= 2.6"
+options:
+    server_url:
+        description:
+            - URL of Foreman server
+        required: true
+    username:
+        description:
+            - Username on Foreman server
+        required: true
+    password:
+        description:
+            - Password for user accessing Foreman server
+        required: true
+    verify_ssl:
+        description:
+            - Verify SSL of the Foreman server
+        default: false
+    name:
+        description:
+            - Name of the lifecycle environment
+        required: true
+    label:
+        description:
+            - Label of the lifecycle environment. This field cannot be updated.
+    description:
+        description:
+            - Description of the lifecycle environment
+    organization:
+        description:
+            - Organization name that the lifecycle environment is in
+        required: true
+    prior:
+        description:
+            - Name of the parent lifecycle environment
+    state:
+        description:
+            - Whether the lifecycle environment should be present or absent on the server
+        required: true
+        choices:
+            - absent
+            - present
+'''
+
+EXAMPLES = '''
+- name: "Add a production lifecycle environment"
+  katello_lifecycle_environment:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://katello.example.com"
+    verify_ssl: false
+    name: "Production"
+    label: "production"
+    organization: "Default Organization"
+    prior: "Library"
+    description: "The production environment"
+    state: "present"
+'''
+
+RETURN = '''# '''
+
+try:
+    from nailgun.entities import (
+        LifecycleEnvironment
+    )
+    HAS_NAILGUN_PACKAGE = True
+except:
+    HAS_NAILGUN_PACKAGE = False
+
+from ansible.module_utils.ansible_nailgun_cement import (
+    create_server,
+    ping_server,
+    find_organization,
+    find_lifecycle_environment,
+    update_fields,
+)
+
+
+def validate_params(module, state, label=None, description=None, prior=None):
+    message = ""
+    if state != 'present':
+        if label is not None:
+            message += "Label cannot be specified if state is present. "
+        if description is not None:
+            message += "Description cannot be specified if state is present. "
+        if prior is not None:
+            message += "Prior cannot be specified if state is present. "
```

Comment:
I don't follow this error message. Why can prior not be specified if state is present?